alerts_backend/python/app.py

```diff
@@ -69,62 +67,65 @@ def create_alert() -> Response:
-    holder.pop("events")
-    holder.pop("max_weekly")
-    # rename dates to avoid sql keyword "end" issue
-    holder["start_date"] = holder.pop("start")
-    holder["end_date"] = holder.pop("end")
-    database_data: Dict[str, Union[str, int]] = holder
-    database_data["fa_alert_id"] = fa_alert_id
-
-    response_frontend["Alert_id"] = fa_alert_id
-    if insert_into_db(database_data) == -1:
-        response_frontend["Description"] = """Database insertion error, 
-        check your database configuration"""
-        return jsonify(response_frontend)
-
-    response_frontend["Success"] = True
-    response_frontend["Description"] = "Request sent successfully"
-    return jsonify(response_frontend)
-
-
-app.run(host="0.0.0.0", port=5000, debug=True)
+        data = request.json
+        api_resource = "/alerts"
+
+        # Check if max_weekly and events in data
+        if "events" not in data:
+            # Assume want all events to be false
+            data["events"] = {
+                "arrival": False,
+                "departure": False,
+                "cancelled": False,
+                "diverted": False,
+                "filed": False,
+            }
+        if "max_weekly" not in data:
+            data["max_weekly"] = 1000
+
+        app.logger.info(f"Making AeroAPI request to POST {api_resource}")
+        result = AEROAPI.post(f"{AEROAPI_BASE_URL}{api_resource}", json=data)
+        if result.status_code != 201:
+            # return to front end the error, decode and clean the response
+            try:
+                processed_json = result.json()
+                r_description = f"Error code {result.status_code} with the following description: {processed_json['detail']}"
+            except json.decoder.JSONDecodeError:
+                r_description = f"Error code {result.status_code} could not be parsed into JSON. The following is the HTML response given: {result.text}"
+        else:
+            # Package created alert and put into database
+            fa_alert_id = int(result.headers["Location"][8:])
+            r_alert_id = fa_alert_id
+            data.pop("events")
+            data.pop("max_weekly")
+            # rename dates to avoid sql keyword "end" issue
+            # default to None in case a user directly submits an incomplete payload
+            data["start_date"] = data.pop("start", None)
+            data["end_date"] = data.pop("end", None)
+            data["fa_alert_id"] = fa_alert_id
+
+            if insert_into_db(data) == -1:
+                r_description = f"Database insertion error, check your database configuration. Alert has still been configured with alert id {r_alert_id}"
+            else:
+                r_success = True
+                r_description = f"Request sent successfully with alert id {r_alert_id}"
+
+    return jsonify({"Alert_id": r_alert_id, "Success": r_success, "Description": r_description})
+
+
+app.run(host="0.0.0.0", port=5001, debug=True)
```

Comment:
I think an accidental port change got through
```suggestion
app.run(host="0.0.0.0", port=5000, debug=True)
```